### PR TITLE
Prevent overlapping jobs

### PR DESCRIPTION
**Pull request by johnduffell**
the admin box in CODE and PROD keeps dying because of a thread leak (gets up to 12000+ threads!) causing memory exhaustion.

The quartz scheduler will happily create threads every time a job is run even if the previous instances are still running.

I have added something firstly to monitor thread count on the boxes to cloudwatch, but also to decline to start a job if the previous one is still running.  I suspect FastlyCloudwatchLoadJob but I need to see it in prod as it's not leaking as quickly locally.

I have made it add a log line when it declines to start a new job, but I am not sure whether we should be notifying ourselves more loudly if a job isn't running.

@gklopper

### CODE REVIEW

File: common/app/common/jobs.scala [hunk 1]
```diff
@@ -9,13 +9,23 @@ import play.api.Play
...
-      f()
+      val name = context.getJobDetail.getKey.getName
+      val f = jobs(name)
+      if (outstanding(name).get() > 0) {
+        log.warn(s"didn't run scheduled job $name because the previous one is still in progress")
+      } else {
+        outstanding.update(name,outstanding(name))
+        outstanding(name).send(_ + 1)
+        f()
+        outstanding(name).send(_ - 1)
```
Comment (gklopper) [inline common/app/common/jobs.scala]: maybe `try { } finally{ }`

Comment (johnduffell) [inline common/app/common/jobs.scala]: good point @gklopper , done that now

File: common/app/common/jobs.scala [hunk 2]
```diff
@@ -11,11 +12,25 @@ import Play.current
 object Jobs extends Logging {
   private val scheduler = StdSchedulerFactory.getDefaultScheduler()
   private val jobs = mutable.Map[String, () => Unit]()
+  private val outstanding = mutable.Map[String,AtomicInteger]().withDefaultValue(new AtomicInteger(0))
```
Comment (gklopper) [inline common/app/common/jobs.scala]: One more thing, this Map is not thread safe. I tend to use the Java ConcurrentHashMap something like this http://stackoverflow.com/a/18667940

**Comment by gklopper** [comment]
Making the map concurrent is very important, but other than that :+1:

**Pull request merged**